src/Init/Data/Nat/Basic.lean

```diff
@@ -370,19 +393,40 @@ theorem le_or_eq_of_le_succ {m n : Nat} (h : m ≤ succ n) : m ≤ n ∨ m = suc
        have : succ m ≤ succ n := succ_le_of_lt this
        Or.inl (le_of_succ_le_succ this))
 
+theorem le_or_eq_of_le_add_one {m n : Nat} (h : m ≤ succ n) : m ≤ n ∨ m = n + 1 :=
```

Comment:
There's still a succ in the hypothesis?

Reply:
Can't you just prove this from `le_or_eq_of_le_succ` rather than repeating the proof?